### PR TITLE
fix(discover) Allow for alternate ISO8601 format

### PR DESCRIPTION
**Pull request by evanh**
Our date filters weren't allowing for the timezone to be in +00:00 format at the
end of the string, which the frontend sometimes adds.

### CODE REVIEW

File: src/sentry/search/utils.py [hunk 1]
```diff
@@ -139,6 +141,8 @@ def parse_datetime_value(value):
     # timezones are not supported and are assumed UTC
     if value[-1:] == "Z":
         value = value[:-1]
+    if len(value) >= 6 and value[-6] == "+":
+        value = value[:-6]
```
Comment (markstory) [inline src/sentry/search/utils.py:145]: Should we consider using multiple timeformats like we do on line 159 or is strptime not able to handle the timezone information.

**Review by markstory — APPROVED** [review]

File: src/sentry/search/utils.py (hunk 1 above)
Comment (evanh) [reply src/sentry/search/utils.py:145]: Hmmm I assumed based on the comment that we were stripping the timezone. On closer inspection that's not what happens.

File: src/sentry/api/event_search.py [hunk 2]
```diff
@@ -140,6 +140,7 @@ def translate(pat):
 quoted_key           = ~r"\"([a-zA-Z0-9_\.:-]+)\""
 
 date_format          = ~r"\d{4}-\d{2}-\d{2}(T\d{2}:\d{2}:\d{2}(\.\d{1,6})?)?Z?(?=\s|$)"
+alt_date_format      = ~r"\d{4}-\d{2}-\d{2}(T\d{2}:\d{2}:\d{2}(\+\d{2}:\d{2})?)?(?=\s|$)"
```
Comment (markstory) [inline src/sentry/api/event_search.py:143]: Could this format also have microseconds in it?

Comment (evanh) [reply src/sentry/api/event_search.py:143]: Hmm yeah according to the spec it can. I'll add that.

**Review by markstory — APPROVED** [review]

**Pull request merged**